examples/dao/setup_sigma_dao.sql

```diff
@@ -18,12 +19,14 @@ CREATE OR REPLACE FUNCTION sigmadao_trigger_fn()
-	app_bytecode text; -- app hash
-	i record;
+	sigma_dao_app_bytecode TEXT := 'BiAFAQAEAgMmGQp2b3RpbmdfZW5kA3llcwdkZXBvc2l0DmV4ZWN1dGVfYmVmb3JlAm5vBHR5cGULbWluX3N1cHBvcnQCaWQMZ292X3Rva2VuX2lkCGV4ZWN1dGVkBGZyb20JcmVjaXBpZW50BmFtb3VudAJwXwdhYnN0YWluA3VybAloYXNoX2FsZ28Mdm90aW5nX3N0YXJ0BmFzYV9pZAxkZXBvc2l0X2xvY2sEbmFtZQh1cmxfaGFzaANtc2cMbWluX2R1cmF0aW9uDG1heF9kdXJhdGlvbjEYIxJABjwxGSQSMRmBBRIRQAYtMRklEjEZIhIRQAYaNhoAgA9vcHRpbl9nb3ZfdG9rZW4SQAXZNhoAgAxhZGRfcHJvcG9zYWwSQASMNhoAgBJkZXBvc2l0X3ZvdGVfdG9rZW4SQARFNhoAgA1yZWdpc3Rlcl92b3RlEkADezYaAIAHZXhlY3V0ZRJAAmw2GgCAFXdpdGhkcmF3X3ZvdGVfZGVwb3NpdBJAAgk2GgCAEWNsZWFyX3ZvdGVfcmVjb3JkEkABJjYaAIAOY2xvc2VfcHJvcG9zYWwSQAABADIHIytiDUAA/zIHIyhiDkAA7zIHIyhiDSMpYicGZA8QIyliIycEYg0QIhJAAM0yByMrYg4yByMoYg0QMgcjKGINIyliJwZkDxAjKWIjJwRiDRAjEhBAAAEAJTUAIyMnB2M1CzUKMgQiEjQLIhIQMSAyAxIxCTIDEhAxFTIDEhAQIycJYiISIytiMgcMESISNAAiEyMoYjIHDBAiEhEQRLEkshAnCGSyETEAshQqZLISI7IBsyMnFGgjJw9oIycVaCMnEGgjJxFoIyhoIytoIycFaCMnCmgjJwtoIycSaCMnDGgjJxZoIycHaCMnCWgjKWgjJwRoIycOaCJDIjUAQv9bIQQ1AEL/VCQ1AEL/TiMjJw02HAFQYzUJNQgyByIrYg1AAK0yByIoYg5AAJ0yByIoYg0iKWInBmQPECIpYiInBGINECISQAB7MgciK2IOMgciKGINEDIHIihiDSIpYicGZA8QIiliIicEYg0QIxIQQAABACU1ADIHIihiDjIHIitiDjQAIhIQIicJYiMSEBFAACslNQEyBCISRDQJIhJAAAojJw02HAFQaCJDNAgiJwdiEjQBIhIQIhJB/+UAIjUBQv/SIjUAQv+tIQQ1AEL/piQ1AEL/oDIEIhIxIDIDEjEJMgMSEDEVMgMSEBAyBzEAJxNiDRBEsSSyECcIZLIRMQCyFDYaAReyEiOyAbMxACojKmI2GgEXCWYiQzIHIitiDUAA8DIHIihiDkAA4DIHIihiDSIpYicGZA8QIiliIicEYg0QIhJAAL4yByIrYg4yByIoYg0QMgciKGINIiliJwZkDxAiKWIiJwRiDRAjEhBAAAEAJTUANAAiEiInCWIjEhBEIicFYiISQABSIicFYiUSQAAXIicFYiEEEkAAAQAyBCISRCInCSJmIkMyBCUSMwEQJBIQMwETIicKYhIQMwEUIicLYhIQMwESIicMYhIQMwERIicSYhIQREL/xzIEJRIzARAiEhAzAQAiJwpiEhAzAQciJwtiEhAzAQgiJwxiEhBEQv+eIjUAQv9qIQQ1AEL/YyQ1AEL/XSMjJw02HAFQYzUJNQgyBCISIicRYjIHDhAyByIoYg4QIypiIw0QRDQJIxJAAHc0CCInB2ITQABeADcAGgEpEkAARzcAGgEnBBJAAC43ABoBJw4SQAABACInDiInDmIjKmIIZiMnE2IiKGIOQAACIkMjJxMiKGJmQv/0IicEIicEYiMqYghmQv/aIikiKWIjKmIIZkL/zSMnDTYcAVAiJwdiZkL/lCMnDTYcAVAiJwdiZkL/hTIEIg8iOBAkEhAiOBEnCGQSECI4FDIKEhAiOBIjDxBEIyojKmIzARIIZiJDIycFYiMSRDIEIg8iOBAkEhAiOBEnCGQSECI4FDIKEhAiOBIjDxBEMwEgMgMSMwESKmQSEEQjJxQ2GgFmIycPNhoCZiMnFTYaA2Y2GgSAABJAANYjJxA2GgRmNhoFFzUCNhoGFzUDNhoHFzUENhoIFzUFNAIyBw1EIycRNAJmNAM0Ag0nF2Q0AzQCCQ4QJxhkNAM0AgkPEEQjKDQDZjQENAMNRCMrNARmNAUiEjQFJRIRNAUhBBIRRCMnBTQFZiMnBWIiEkAASSMnBWIlEkAAHyMnBWIhBBJAAAEAIycWNhoJZiMnBzEXZiMnCSNmIkMjJwo2GglmIycSNhoKF2YjJws2GgtmIycMNhoMF2ZC/9IjJwo2GglmIycLNhoKZiMnDDYaCxdmQv+5IycQgAZzaGEyNTZmQv8iMgQiEjEgMgMSEDYwACcIZBIQRLEkshAnCGSyETIKshQjshIjsgGzIkMyBCISRCJDI0M2GgIXIw02GgIXNhoDFwwQNhoGF3EANQc1BjQHEEQqNhoAF2cnBjYaARdnJxc2GgIXZycYNhoDF2cnDzYaBGeACGRhb19uYW1lNhoFZycINhoGF2ciQw==';
+	app_bytecode TEXT; -- app hash
+	dao_name_key_bytecode CHAR(255) := 'ZGFvX25hbWU='; -- Byte code of 'dao_name' key
+	dao_name CHAR(255); -- dao_name
+	i RECORD;
 BEGIN
 	/* 	Check to avoid duplicate entries in table. id attribute increments despite pass/fail insertion  */
 	IF EXISTS (SELECT FROM sigma_daos p WHERE p.app_id = NEW.asset) THEN
```

Comment:
how and when this is possible?

Reply:
txn relation will have two entries for same app id and data, that's why our trigger will be called two times. This is just to avoid duplicate entries in table. This is safe check to avoid duplicates. Also, `id` attributes will increase even if insertion failed on 2nd time,

Reply:
Could you this to the comment? What are the two entries?

Reply:
ok. I have added the comment.